### ISSUE TITLE
Video mp4 download

### ISSUE THREAD

**Issue by marcoscale98** [C0D3D3V/Moodle-Downloader-2#32]
It would be nice to introduce the possibility of downloading videos in mp4 format. Currently the system only produces a link which, once opened, downloads the video. Maybe you could choose it during configuration

**Comment by C0D3D3V** [comment]
The same feature request already exists: #12

**Comment by C0D3D3V** [comment]
The problem is that these links are all external links and there is no perfect way to download them. 
The idea would be to collect a list of external links, categorize them and then download them with third party tools like youtube-dl or jdownloader.
But there is no way to determine if it is a video, document or anything else.

**Comment by marcoscale98** [comment]
I am referring to internal links. I managed to make a spartan patch (all links are transformed into mp4)

**Comment by C0D3D3V** [comment]
Can you send a screenshot and the source code from the Moodle site where the mp4 is not downloaded. Normally all files stored on moodle will be downloaded.

**Comment by marcoscale98** [comment]
> Can you send a screenshot and the source code from the Moodle site where the mp4 is not downloaded. Normally all files stored on moodle will be downloaded.

![image](https://user-images.githubusercontent.com/43850400/82128181-7374cd80-97b9-11ea-88c6-3c91753462ac.png)
For example, don't download "first part.mp4" but "first part.desktop"

https://pastebin.com/nsX4DFs3
This is the source code of the page

**Comment by C0D3D3V** [comment]
Thank you :) I have looked at the source code.
 That's exactly what Issue #12 is about. The mp4 file was not uploaded directly to moodle but only given as URL. You can see that by right clicking on the link and then clicking "open in a new tab". So far i don't have a good solution for this problem. I could follow the link and just download the file. But behind the link could be a website. 
Maybe you have an idea. 

I  could let the user define which links should be downloaded automatically with which program.  For example, if the link links to a youtube video that is then downloaded using youtube-dl.

**Comment by marcoscale98** [comment]
> Maybe you have an idea.

I used this method which only worked for the course in question. Instead of creating a link, I created an mp4 file. It works because in the link there is only one video file, if there are more than one it doesn't work.

![image](https://user-images.githubusercontent.com/43850400/82139798-c5097080-982a-11ea-8fd9-9d37b18c8776.png)
In my case, however, the video is in an external link but is visible from moodle
